crates/containerd-shim-wasmtime/src/http_proxy.rs

```diff
@@ -0,0 +1,182 @@
+// Heavily inspired by wasmtime serve command:
+// https://github.com/bytecodealliance/wasmtime/blob/main/src/commands/serve.rs
+
+use std::net::{IpAddr, SocketAddr};
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+
+use anyhow::{bail, Result};
+use wasmtime::component::ResourceTable;
+use wasmtime::Store;
+use wasmtime_wasi::{self as wasi_preview2};
+use wasmtime_wasi_http::bindings::http::types::Scheme;
+use wasmtime_wasi_http::bindings::ProxyPre;
+use wasmtime_wasi_http::body::HyperOutgoingBody;
+use wasmtime_wasi_http::io::TokioIo;
+use wasmtime_wasi_http::{WasiHttpCtx, WasiHttpView};
+
+use crate::instance::WasiCtx;
+
+const DEFAULT_ADDR: SocketAddr =
+    SocketAddr::new(IpAddr::V4(std::net::Ipv4Addr::new(0, 0, 0, 0)), 8080);
+
+const DEFAULT_BACKLOG: u32 = 100;
+
+fn wasi_store_for_request(handler: &ProxyHandlerInner, req_id: u64) -> Store<WasiCtx> {
+    let engine = handler.instance_pre.engine();
+    let mut builder = wasi_preview2::WasiCtxBuilder::new();
+
+    builder.envs(&handler.env);
+    builder.env("REQUEST_ID", req_id.to_string());
+
+    let ctx = WasiCtx {
+        wasi_preview1: None,
+        wasi_preview2_cli: builder.build(),
+        wasi_preview2_http: WasiHttpCtx::new(),
+        resource_table: ResourceTable::default(),
+        envs: vec![],
```

Comment:
question: what is this envs used for?

Reply:
That was some leftover. It is now removed.